### DISCUSSION TITLE
How to use Index signatures as types

### DISCUSSION THREAD

**Discussion by lantos1618** [MichalLytek/type-graphql#1303] [Q&A] [answered]
Hi, I've been looking on stacker overflow, the typegraphql and on the gitter chat on how implement types with index signatures.
Found it was mentioned here. https://gitter.im/type-graphql/Lobby?at=5f002ad4d65a3b0292c92593 But not a clear way on how to acheive this or if it should be acheived. 

How can you use index signatures as types?
```ts
import { ObjectType } from "type-graphql";
import { GraphQLJSONObject } from "graphql-type-json";

@ObjectType()
export class Label {
    @Field(of => GraphQLJSONObject, { nullable: true })
    [label: string]: string
}
```
> error:Decorators are not valid here.ts(1206)

**Comment by MichalLytek** [comment, marked as answer]
You need to represent the object with dynamic keys as an array of key-value objects.

```ts
@ObjectType()
export class LabelValue {  
	@Field()
	label: string;
	
	 @Field()
    value: string;
}

@ObjectType()
export class Foo {    
	@Field(() => [LabelValue])
	labels: LabelValue[];
}
```
